content/en/apps/guides/hosting/monitoring/postgres-ingest.md

```diff
@@ -0,0 +1,437 @@
+
+```yaml
+replication_failure_reasons:
+  query: |
+    SELECT 
+      metric as failure_type,
+      sum(count) AS "count"
+    FROM 
+      public.app_monitoring_replication_failure_reasons
+    WHERE 
+      partner_name IN ('partner_name_here')
+    GROUP BY 
+      failure_type
+    ORDER BY 
+      count DESC
+  metrics:
+    - failure:
+        usage: "LABEL"
+        description: "Name of the failure"
+    - sequence:
+        usage: "TABLE"
+        description: "Replication failure reasons"
+```
+
+Note that you can use any value on the first line, seen as  `replication_failure_reasons` above.  It will be the field name we'll use in the 4th step below when exploring the new data in Grafana.
+
+### Add new Postgres Exporter
+
+In a new file  `~/extra-sql-compose.yml`, define your new Postgres exporter as well as add a mount to the existing Grafana and Prometheus services. Note that the `DATA_SOURCE_NAME` value will need to have the following variables added to you `.env` file you created to deploy Watchdog:
+
+* `EXTRA_SQL_USER` - Postgres user to use when logging in
+* `EXTRA_SQL_PASS` - Password for `EXTRA_SQL_USER` above
+* `EXTRA_SQL_SERVER` - URL or IP for your Postgres server
+* `EXTRA_SQL_PORT` - Port of server, defaults to `5432` it not declared.
+* `EXTRA_SQL_DATABASE` - Actual string of database name (eg `extra_monitoring` or `health_stats`), will be different for each install.
+
+```yaml
+services:
+
+  prometheus:
+    volumes:
+      - ./extra-sql-prometheus.yml:/etc/prometheus/scrape_configs/extra-sql-prometheus.yml:ro
+
+  grafana:
+    volumes:
+      - ./extra-sql-dashboard.json:/etc/grafana/provisioning/dashboards/CHT/cht_admin_extra_sql.json:ro
+
+  extra_sql_exporter:
+    image: prometheuscommunity/postgres-exporter:latest
+    command:
+      # disables the collection of all metrics except for custom queries 
+      - '--no-collector.database'
+      - '--no-collector.postmaster'
+      - '--no-collector.process_idle'
+      - '--no-collector.replication'
+      - '--no-collector.replication_slot'
+      - '--no-collector.stat_bgwriter'
+      - '--no-collector.stat_database'
+      - '--no-collector.statio_user_tables'
+      - '--no-collector.stat_statements'
+      - '--no-collector.stat_user_tables'
+      - '--disable-default-metrics'
+      - '--disable-settings-metrics'
+    volumes:
+      - ./extra-sql-queries.yml:/extra-sql-queries.yml
+    environment:
+      DATA_SOURCE_NAME: "postgresql://${EXTRA_SQL_USER:-NO DB USER SPECIFIED}:${EXTRA_SQL_PASS:-NO DB PASSWORD SPECIFIED}@${EXTRA_SQL_SERVER:-.NO DB SERVER SPECIFIED}:${EXTRA_SQL_PORT:-5432}/${EXTRA_SQL_DATABASE:-.NO DB SPECIFIED}?sslmode=disable"
+      PG_EXPORTER_EXTEND_QUERY_PATH: "/extra-sql-queries.yml"
+    restart: always
+    networks:
+     - cht-watchdog-net
+```
+
+If there's a missing mount target, then Docker will error.  Avoid the error by creating an empty file called `~/extra-sql-dashboard.json`.  We'll use this in the last step below
```

Comment:
```suggestion
```

Is there a reason we are defining the docker-compose config in the instructions before setting up the scrape config or the dashboard?  To me, it is more confusing to be trying to map "future" files in the compose.yml vs just creating each of the config files up-front and then defining the docker-compose file...

Reply:
Yeah - it is a bit confusing, agreed. 

My thinking was that I wanted one edit to the compose file.  Once you had that in place, you could actually prototype the query, create the dashboard and save the JSON out to paste in the file. Seemed clean-ish.

Otherwise you have to:
1. set up the scrape and the query file, do most of the compose.
2. prototype the query, create the dashboard and save the JSON
3. create the JSON file
4. update the compose file to have JSON file mounted in grafana
5. restart and test it worked

so I'm shaving off a few steps as they felt dirty-ish.

I'll lightly push back but if you still disagree, I'll add more steps to the docs!